rust_src/src/buffers.rs

```diff
@@ -103,3 +103,12 @@ pub fn get_buffer(buffer_or_name: LispObject) -> LispObject {
         ))
     }
 }
+
+/// Return the current buffer as a Lisp object.
+#[lisp_fn]
+pub fn current_buffer() -> LispObject {
```

Comment:
@birkenfeld I've added a "from_ptr" equivalent in https://github.com/Wilfred/remacs/pull/251, and was planning on changing this call site to use that if this PR gets merged first.